src/filters/prefix-tree.ts

```diff
@@ -12,48 +15,86 @@
  * However, we have to maximize performance and remove any overhead that third-party packages might have (to provide
  * more functionality that is not needed here).
  */
-export class PrefixTree {
+export class PrefixTree extends IndexFilter{
     private root : PTElement;
 
     /**
      * Create a PrefixTree and add initialize it with a set of terms (optional).
-     * @param terms
+     * @param tokens
      */
-    constructor(terms? : string[]) {
+    constructor(tokens? : string[]) {
+        super();
         this.root = new PTNode(); // not a leaf! we do not want to match any string!
-
-        if (terms) { // add provided terms if any
-            for (let term of terms) {
-                this.addTermToTree(term);
-            }
-        }
+        if (tokens) { super.addSearchTerms(tokens); }
```

Comment:
`super` will call the `addSearchTerms` in `Filter` 
-> `for (let token of tokens)` 
Does it break if `tokens` is `undefined`?